### DISCUSSION TITLE
Need help transferring Selenium code to SeleniumBase

### DISCUSSION THREAD

**Discussion by hardened3348** [seleniumbase/SeleniumBase#2706] [Q&A] [answered]
Hi there,

I'm currently trying to adapt the following code from Selenium to SeleniumBase:
```
PROFILE_ID = 'PROFILE_UUID'
WEBDRIVERPATH = Service(
    executable_path=r'./chromedriver/chromedriver-win64/chromedriver.exe')
LOCAL_API = 'http://localhost:58888/api/profiles'

def get_webdriver(port):
    chrome_options = Options()
    chrome_options.add_experimental_option(
        'debuggerAddress', f'127.0.0.1:{port}')
    driver = webdriver.Chrome(service=WEBDRIVERPATH, options=chrome_options)
    return driver

def get_debug_port(profile_id):
    data = requests.post(
        f'{LOCAL_API}/start', json={'uuid': profile_id, 'headless': False, 'debug_port': True}
    ).json()
    return data['debug_port']

def main():
    port = get_debug_port(PROFILE_ID)
    driver = get_webdriver(port)
    driver.get('http://example.com')
```
I've been struggling to make it work with SeleniumBase, specifically with undetected mode enabled. Could you please help me figure out what adjustments I need to make? Thanks a lot!

I've tried using this command with SeleniumBase, but it doesn't seem to be working:

`with SB(uc=True, chromium_arg=f'--remote-debugging-address=127.0.0.1,--remote-debugging-port={port}') as driver:`
Any assistance would be greatly appreciated!

**Comment by mdmintz** [comment, marked as answer]
The debugger address and remote debugging port are already set by UC Mode, so you can't configure that.
The driver for UC Mode must be `uc_driver` / `uc_driver.exe`, and that is already set, so you can't configure that either.

All you need is:

```python
from seleniumbase import SB

with SB(uc=True) as sb:
    sb.driver.uc_open_with_reconnect("URL")
```

or 

```python
from seleniumbase import Driver

driver = Driver(uc=True)
driver.uc_open_with_reconnect("URL")

# ...

driver.quit()
```

**Reply by hardened3348** [reply]
Thank you for your message.

To use the specific profile launched with the debug port, I need to utilize OctoBrowser. This is based on the documentation provided [here](https://documenter.getpostman.com/view/1801428/UVC6i6eA#f32c145e-7327-4f88-89ad-e37e21f8eb40).
After launching it, there is no way to connect to it with seleniumbase?

If you don't use OctoBrowser, you won't be able to launch that exact profile.

**Comment by hardened3348** [comment]
Fixed 
`with SB(uc=True, chromium_arg=f'--remote-debugging-address=127.0.0.1,--remote-debugging-port={port}') as driver:`

checked and the debugging address is 0.0.0.0 by default

After launching the profile, getting the port, this will work:
`with SB(uc=True, chromium_arg=f'--remote-debugging-address=0.0.0.0,--remote-debugging-port={port}') as driver:`

Thanks again for your help!
